templates/typescript_gapic/_util.njk

```diff
@@ -301,6 +303,32 @@ request.{{ oneComment.paramName.toCamelCase() }}
 {%- endfor %}
     });
 {%- endif %}
+{%- endif %}
+{#-If dynamic routing annotation exists and is non empty, then send dynamic routing header-#}
+{%- if method.sendHeaderRequestParams and method.useDynamicRoutingHeader -%}    
+    options.otherArgs = options.otherArgs || {};
+    options.otherArgs.headers = options.otherArgs.headers || {};
```

Comment:
These are common code for 3 situation, so we could take it out.

Reply:
For these 3 situation, we could simplify as 
```
options.otherArgs = options.otherArgs || {};
options.otherArgs.headers = options.otherArgs.headers || {};
{%- if mehtod.dynamicRoutingRequestParams.length > 0 -%} {
	// send dynamic  routing headers
}
{%- elseif method.headerRequestParams.length > 0 -%} {
	// sent regular headers
```
And in the `protos.ts`, we could do 
```
if (!methodDynamicRouting) {
	method.headerRequestParams = getHeaderRequestParams(
      method.options?.['.google.api.http']
    );
} else if (methodDynamicRoutingParameters && methodDynamicRoutingParameters.length > 0) {
	method.dynamicRoutingRequestParams = getDynamicHeaderRequestParams(
      methodDynamicRoutingParameters
    );
}
```
We could only use two variables, `headerRequestParams` and `dynamicRoutingRequestParams` to check the conditions. Either `headerRequestParams length > 0` sent regular headers or  `dynamicRoutingRequestParams length > 0` sent dynamic headers. If both variable equals to 0, then do no sent the headers.

Reply:
Ping me anytime if the logic confuse you. :)